tests/api/v2/handlers/test_agents_api.py

```diff
@@ -0,0 +1,222 @@
+        resp = await api_v2_client.get('/api/v2/agents/123')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_nonexistent_get_agent_by_id(self, api_v2_client, api_cookies):
+        resp = await api_v2_client.get('/api/v2/agents/999', cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_create_agent(self, api_v2_client, api_cookies, new_agent_payload):
+        resp = await api_v2_client.post('/api/v2/agents', cookies=api_cookies, json=new_agent_payload)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
+        assert agent_dict['sleep_min'] == new_agent_payload['sleep_min']
+        assert agent_dict['sleep_max'] == new_agent_payload['sleep_max']
+        assert agent_dict['paw'] == new_agent_payload['paw']
+        assert agent_dict['executors'] == new_agent_payload['executors']
+
+    async def test_unauthorized_create_agent(self, api_v2_client, new_agent_payload):
+        resp = await api_v2_client.post('/api/v2/agents', json=new_agent_payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_update_agent(self, api_v2_client, api_cookies, test_agent, updated_agent_payload):
+        resp = await api_v2_client.patch('/api/v2/agents/123', cookies=api_cookies, json=updated_agent_payload)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
+        assert agent_dict == updated_agent_payload
+
+    async def test_unauthorized_update_agent(self, api_v2_client, test_agent, updated_agent_payload):
+        resp = await api_v2_client.patch('/api/v2/agents/123', json=updated_agent_payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_nonexistent_update_agent(self, api_v2_client, api_cookies, updated_agent_payload):
+        resp = await api_v2_client.patch('/api/v2/agents/999', cookies=api_cookies, json=updated_agent_payload)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_create_or_update_existing_agent(self, api_v2_client, api_cookies, test_agent, updated_agent_payload):
+        resp = await api_v2_client.put('/api/v2/agents/123', cookies=api_cookies, json=updated_agent_payload)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
+        assert agent_dict == updated_agent_payload
+
+    async def test_unauthorized_create_or_update_agent(self, api_v2_client, test_agent, updated_agent_payload):
+        resp = await api_v2_client.put('/api/v2/agents/123', json=updated_agent_payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_create_or_update_nonexistent_agent(self, api_v2_client, api_cookies, new_agent_payload):
+        resp = await api_v2_client.put('/api/v2/agents/456', cookies=api_cookies, json=new_agent_payload)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
```

Comment:
this would create a new agent, right? Could we compare agent_dict with new_agent_payload?

Reply:
Yep, this would create a new agent! I'll update some of these fixtures so that we can just compare the dumped objects